framework/src/main/java/org/checkerframework/framework/util/CheckerMain.java

```diff
@@ -110,11 +124,24 @@ public CheckerMain(final File checkerJar, final List<String> args) {
         this.javacJar =
                 extractFileArg(PluginUtil.JAVAC_PATH_OPT, new File(searchPath, "javac.jar"), args);
 
-        final String jdkJarName = PluginUtil.getJdkJarName();
-        this.jdkJar =
-                extractFileArg(PluginUtil.JDK_PATH_OPT, new File(searchPath, jdkJarName), args);
+        if (PluginUtil.getJreVersion() > 8) {
+            final File annotatedJDKSearchPath = new File(searchPath, "annotatedJDK");
+            final int jreVersion = PluginUtil.getJreVersion();
+            final String jdkVersionFolderName = "jdk" + jreVersion;
+            final File jdkVersionModuleLocation =
+                    new File(annotatedJDKSearchPath, jdkVersionFolderName);
+            this.jdkModulesPath =
+                    extractFileArg(PluginUtil.JDK_PATH_OPT, jdkVersionModuleLocation, args);
+            this.jdkJar = null;
+            this.compilationBootclasspath = null;
```

Comment:
Please document the changed meaning of `compilationBootclasspath` and mark that field as `@Nullable`.

Reply:
Done.